### ISSUE TITLE
flush/redraw chart? old chart get updated instead of redrawing

### ISSUE THREAD

**Issue by kunKun-tx** [jettro/c3-angular-directive#61]
I have two donut charts and i use ``on-click-data`` event from the first donut to select and feed new data to the second chart. The data is pushed by a function with $timeout attribute.

It works great with the old version, ``c3-angular - v0.5.1 - 2015-06-07``. But in the newer version, and even in the latest version, it acts strange.  When i invoke the ``on-click-data`` event, instead of re-initializing the second donut, it adds new data to the chart while keeping all the old data.

There are array-emptying commands first line in my ``on-click-data`` event function. It still works. You can see (screenshots below) chart data arrays are clean.

![1](https://cloud.githubusercontent.com/assets/12914967/9728990/8ed4e4c2-55d1-11e5-80b3-3798841ce065.png)
![2](https://cloud.githubusercontent.com/assets/12914967/9729184/3d676036-55d3-11e5-8ac7-d172642a9566.png)

**Comment by jettro** [comment]
Is it possible to give a working sample? I'll try to create one myself but having something to test it with would be nice

**Comment by eiclkun** [comment]
will do that tonight. hopefully i can solve this while minimizing it.

**Comment by eiclkun** [comment]
sorry for the delay, life is hard...
here is the code pen i just put up,
http://codepen.io/anon/pen/LpGaqX
some hacks here and there but you get the idea:
click the **green** part in the first donut chart will clean up the data column and re-inject with new data.

notice that current setting with latest c3-angular will not *flush* chart while adding the new data while  the old ver. will.
(below if the old ver. i was using)
http://cdn.rawgit.com/jettro/c3-angular-directive/d2ec14eb64d42e5cfd06958dc13f2f62ef556c77/c3-angular.min.js

**Comment by wester97** [comment]
Is it possible to use this with a filter?  I have data that I'd like to filter by date range, for instance.

**Comment by kunKun-tx** [comment]
@wester97  you can always filter the data you push into the chartDataColumn

**Comment by jettro** [comment]
Hi @kunKun-tx I finally managed to fix this. Had to do with an optimisation of loading changes instead of redrawing the whole chart. But it seems the data points are added to the points already available. So if they have a different id, they are added. Now I explicitly remove all points before adding the new points and now everything works. I have added a sample with the pie charts in the callback example.
